articles/prototyping-machine-learning-models-with-streamlit/index.md

```diff
@@ -0,0 +1,212 @@
+```
+ 
+Nowe for us to create the virtual environment we use the code below this to isolate the environment.
+ 
+```python
+$ python3.8 -m venv env
+$ source env/bin/activate
+```
+ 
+The first line of code creates the virtual environment while the second activate it. Now copy the dataset as it is in the folder and paste it into this textanalyzer directory, and then create a file and save it as sentiment_analyzer.py, this will be where we will be writing our code to build the streamlit app. The next thing will be to import all the packages we will be using for this tutorial.
+ 
+```python
+import streamlit as st
+from sklearn.feature_extraction.text import CountVectorizer
+from sklearn.naive_bayes import BernoulliNB
+from sklearn.metrics import confusion_matrix
+import numpy as np
+import itertools
+import matplotlib.pyplot as plt
+```
+ 
+After that let’s add a header for our web app and a subheader and to do that we use the code below
+ 
+```python
+st.title("Sentiment Analyzer Based On Text Analysis ")
+st.subheader("Paras Patidar - MLAIT")
+st.write('\n\n')
+```
+ 
+The next step will be to write a function that will get all the data from our directory and bring them up, your directory should look like
+ 
+![Simple](https://github.com/jamessandy/engineering-education/blob/new-article/articles/prototyping-machine-learning-models-with-streamlit/img1.jpg)
+ 
+```python
+@st.cache
+def get_all_data():
+ root = "Datasets/"
+ with open(root + "imdb_labelled.txt", "r") as text_file:
+     data = text_file.read().split('\n')
+ with open(root + "amazon_cells_labelled.txt", "r") as text_file:
+     data += text_file.read().split('\n')
+ with open(root + "yelp_labelled.txt", "r") as text_file:
+     data += text_file.read().split('\n')
+ return data
+```
+ 
+If you notice in the function above the first line is *@st.cache*, this mark tells streamlit that whenever the function is called that there are some certain requirements it should check for and they should be fulfilled, now create a variable called all_data which should call the *get_all_data* function we wrote.
+ 
+```python
+all_data = get_all_data()
+```
+ 
+Next, we create a checkbox using the *st.checkbox* function that will display the dataset, and also the checkbox should have the values from the *all_data* variable we created earlier.
+ 
+```python
+if st.checkbox('Show Dataset'):
+ st.write(all_data)
+```
+ 
+Now we write another function that will carry out data preprocessing and to achieve that we use the code below.
+ 
+```python
+@st.cache
+def preprocessing_data(data):
+ processing_data = []
+ for single_data in data:
+     if len(single_data.split("\t")) == 2 and single_data.split("\t")[1] != "":
+         processing_data.append(single_data.split("\t"))
+ return processing_data
+ 
+```
+ 
+Now let’s create another checkbox that will display *show_preprocessed_data* and will get the data from the *all_data* function and pass it through the preprocessing data function we wrote above.
```

Comment:
Instead of using multiple _and(s)_, use periods. Break it down into simpler sentences.